MSVC build seems to be broken in zsock.c

Due to latest changes (i.e. 'Problem: zsock_send can't send 32/64 bit integers safely') MSVC 2012 build seems to be broken.
```
1>..\..\..\..\src\zsock.c(591): error C2146: syntax error : missing ')' before identifier 'PRIu8'
1>..\..\..\..\src\zsock.c(591): error C2059: syntax error : ')'
1>..\..\..\..\src\zsock.c(594): error C2146: syntax error : missing ')' before identifier 'PRIu16'
1>..\..\..\..\src\zsock.c(594): error C2059: syntax error : ')'
1>..\..\..\..\src\zsock.c(597): error C2146: syntax error : missing ')' before identifier 'PRIu32'
1>..\..\..\..\src\zsock.c(597): error C2059: syntax error : ')'
1>..\..\..\..\src\zsock.c(600): error C2146: syntax error : missing ')' before identifier 'PRIu64'
1>..\..\..\..\src\zsock.c(600): error C2059: syntax error : ')'
1>..\..\..\..\src\zsock.c(751): error C3861: 'atoll': identifier not found
```

See if this helps: https://github.com/zeromq/czmq/pull/884

On Thu, Jan 8, 2015 at 2:31 PM, ttuna <notifications@github.com> wrote:

> Due to latest changes (i.e. 'Problem: zsock_send can't send 32/64 bit
> integers safely') MSVC 2012 build seems to be broken.
>
> 1>..\..\..\..\src\zsock.c(591): error C2146: syntax error : missing ')' before identifier 'PRIu8'
> 1>..\..\..\..\src\zsock.c(591): error C2059: syntax error : ')'
> 1>..\..\..\..\src\zsock.c(594): error C2146: syntax error : missing ')' before identifier 'PRIu16'
> 1>..\..\..\..\src\zsock.c(594): error C2059: syntax error : ')'
> 1>..\..\..\..\src\zsock.c(597): error C2146: syntax error : missing ')' before identifier 'PRIu32'
> 1>..\..\..\..\src\zsock.c(597): error C2059: syntax error : ')'
> 1>..\..\..\..\src\zsock.c(600): error C2146: syntax error : missing ')' before identifier 'PRIu64'
> 1>..\..\..\..\src\zsock.c(600): error C2059: syntax error : ')'
> 1>..\..\..\..\src\zsock.c(751): error C3861: 'atoll': identifier not found
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/czmq/issues/883>.
>

--- a/include/czmq_prelude.h
+++ b/include/czmq_prelude.h
@@ -422,6 +422,7 @@ typedef struct sockaddr_in inaddr_t;    //  Internet socket address structure
 #       define inline __inline
 #   endif
 #   define strtoull _strtoui64
+#   define atoll _atoi64
 #   define srandom srand
 #   define TIMEZONE _timezone
 #   if (!defined (__MINGW32__))

